### PR TITLE
feat(TaskProcessing): Implement enums and default values

### PR DESCRIPTION
**Pull request by marcelklehr**
## Summary
This adds Enum types to input and outputs of Tasks and allows providers to specify the values of those enums, as well as default values for any input slots.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by julien-nc — APPROVED** [review]
:rocket: :boom: :+1: :heart:

**Review by provokateurin — APPROVED** [review]
Quick, feature freeze is tomorrow :D

File: lib/private/TaskProcessing/Manager.php [hunk 1]
```diff
@@ -439,35 +511,64 @@ private function _getTaskTypes(): array {
...
 
+	/**
+	 * Takes task input data and replaces fileIds with File objects
+	 *
+	 * @param array<array-key, list<numeric|string>|numeric|string> $input
+	 * @param array<array-key, numeric|string> ...$defaultSpecs the specs
+	 * @return array<array-key, list<numeric|string>|numeric|string>
+	 */
+	public function fillInputDefaults(array $input, ...$defaultSpecs): array {
+		$spec = array_reduce($defaultSpecs, fn ($carry, $spec) => $carry + $spec, []);
```
Comment (julien-nc) [inline lib/private/TaskProcessing/Manager.php:559]: This always writes default values even when a value is set in the submitted input.

**Review by julien-nc — CHANGES_REQUESTED** [review]

File: lib/private/TaskProcessing/Manager.php [hunk 2]
```diff
@@ -439,35 +511,64 @@ private function _getTaskTypes(): array {
...
+		return $spec + $input;
```
Comment (julien-nc) [inline lib/private/TaskProcessing/Manager.php:560]:
```suggestion
		$spec = array_reduce($defaultSpecs, fn ($carry, $spec) => array_merge($carry, $spec), []);
		return array_merge($spec, $input);
```

This always writes default values even when a value is set in the submitted input.

**Review by julien-nc — CHANGES_REQUESTED** [review]

**Review by julien-nc — APPROVED** [review]

**Review by marcelklehr — COMMENTED** [review]
I approve

**Comment by marcelklehr** [comment]
Woop wooop

**Pull request merged**